lua_modules/dht22/dht22.lua

```diff
@@ -0,0 +1,101 @@
+-- ***************************************************************************
+-- DHT22 module for ESP8266 with nodeMCU
+--
+-- Written by Javier Yanez 
+-- but based on a script of Pigs Fly from ESP8266.com forum
+--
+-- MIT license, http://opensource.org/licenses/MIT
+-- ***************************************************************************
+
+local moduleName = ...
+local M = {}
+_G[moduleName] = M
+
+local humidity
+local temperature
+local checksum
+local checksumTest
+
+function M.read(pin)
+  humidity = 0
+  temperature = 0
+  checksum = 0
+
+  -- Use Markus Gritsch trick to speed up read/write on GPIO
+  gpio_read = gpio.read
+  gpio_write = gpio.write
```

Comment:
Well, https://github.com/nodemcu/nodemcu-firmware/blob/master/lua_modules/ds18b20/ds18b20.lua#L124 assigns a global called `read`. Please, confirm. This is hardly what you wanted, as it may clash with others.
Please, consider enclosing private stuff in `do ... end` and explicit exposure, as I've written below.